providers/fips/self_test_data.inc

```diff
@@ -284,8 +286,207 @@ static const ST_KAT_PARAM sskdf_params[] = {
+    0x3c, 0x69, 0x62, 0x26, 0x65, 0x0a, 0x86, 0x87,
+};
+static int pbkdf2_iterations = 4096;
+static int pbkdf2_pkcs5 = 1;
+static const ST_KAT_PARAM pbkdf2_params[] = {
+    ST_KAT_PARAM_UTF8STRING(OSSL_KDF_PARAM_DIGEST, pbkdf2_digest),
+    ST_KAT_PARAM_OCTET(OSSL_KDF_PARAM_PASSWORD, pbkdf2_password),
+    ST_KAT_PARAM_OCTET(OSSL_KDF_PARAM_SALT, pbkdf2_salt),
+    ST_KAT_PARAM_INT(OSSL_KDF_PARAM_ITER, pbkdf2_iterations),
+    ST_KAT_PARAM_INT(OSSL_KDF_PARAM_PKCS5, pbkdf2_pkcs5),
+    ST_KAT_PARAM_END()
+};
+
+static const char sshkdf_digest[] = "SHA1";
+static int sshkdf_type = EVP_KDF_SSHKDF_TYPE_INITIAL_IV_CLI_TO_SRV;
+static const unsigned char sshkdf_key[] = {
+    0x00, 0x00, 0x00, 0x80, 0x55, 0xba, 0xe9, 0x31,
+    0xc0, 0x7f, 0xd8, 0x24, 0xbf, 0x10, 0xad, 0xd1,
+    0x90, 0x2b, 0x6f, 0xbc, 0x7c, 0x66, 0x53, 0x47,
+    0x38, 0x34, 0x98, 0xa6, 0x86, 0x92, 0x9f, 0xf5,
+    0xa2, 0x5f, 0x8e, 0x40, 0xcb, 0x66, 0x45, 0xea,
+    0x81, 0x4f, 0xb1, 0xa5, 0xe0, 0xa1, 0x1f, 0x85,
+    0x2f, 0x86, 0x25, 0x56, 0x41, 0xe5, 0xed, 0x98,
+    0x6e, 0x83, 0xa7, 0x8b, 0xc8, 0x26, 0x94, 0x80,
+    0xea, 0xc0, 0xb0, 0xdf, 0xd7, 0x70, 0xca, 0xb9,
+    0x2e, 0x7a, 0x28, 0xdd, 0x87, 0xff, 0x45, 0x24,
+    0x66, 0xd6, 0xae, 0x86, 0x7c, 0xea, 0xd6, 0x3b,
+    0x36, 0x6b, 0x1c, 0x28, 0x6e, 0x6c, 0x48, 0x11,
+    0xa9, 0xf1, 0x4c, 0x27, 0xae, 0xa1, 0x4c, 0x51,
+    0x71, 0xd4, 0x9b, 0x78, 0xc0, 0x6e, 0x37, 0x35,
+    0xd3, 0x6e, 0x6a, 0x3b, 0xe3, 0x21, 0xdd, 0x5f,
+    0xc8, 0x23, 0x08, 0xf3, 0x4e, 0xe1, 0xcb, 0x17,
+    0xfb, 0xa9, 0x4a, 0x59,
+};
+static const unsigned char sshkdf_xcghash[] = {
+    0xa4, 0xeb, 0xd4, 0x59, 0x34, 0xf5, 0x67, 0x92,
+    0xb5, 0x11, 0x2d, 0xcd, 0x75, 0xa1, 0x07, 0x5f,
+    0xdc, 0x88, 0x92, 0x45,
+};
+static const unsigned char sshkdf_session_id[] = {
+    0xa4, 0xeb, 0xd4, 0x59, 0x34, 0xf5, 0x67, 0x92,
+    0xb5, 0x11, 0x2d, 0xcd, 0x75, 0xa1, 0x07, 0x5f,
+    0xdc, 0x88, 0x92, 0x45,
+};
+static const unsigned char sshkdf_expected[] = {
+    0xe2, 0xf6, 0x27, 0xc0, 0xb4, 0x3f, 0x1a, 0xc1,
+};
+static const ST_KAT_PARAM sshkdf_params[] = {
+    ST_KAT_PARAM_UTF8STRING(OSSL_KDF_PARAM_DIGEST, sshkdf_digest),
+    ST_KAT_PARAM_INT(OSSL_KDF_PARAM_SSHKDF_TYPE, sshkdf_type),
+    ST_KAT_PARAM_OCTET(OSSL_KDF_PARAM_KEY, sshkdf_key),
+    ST_KAT_PARAM_OCTET(OSSL_KDF_PARAM_SSHKDF_XCGHASH, sshkdf_xcghash),
+    ST_KAT_PARAM_OCTET(OSSL_KDF_PARAM_SSHKDF_SESSION_ID, sshkdf_session_id),
+    ST_KAT_PARAM_END()
+};
+
+static const char tls12prf_digest[] = "SHA256";
+static const unsigned char tls12prf_secret[] = {
+    0x20, 0x2c, 0x88, 0xc0, 0x0f, 0x84, 0xa1, 0x7a,
+    0x20, 0x02, 0x70, 0x79, 0x60, 0x47, 0x87, 0x46,
+    0x11, 0x76, 0x45, 0x55, 0x39, 0xe7, 0x05, 0xbe,
+    0x73, 0x08, 0x90, 0x60, 0x2c, 0x28, 0x9a, 0x50,
+    0x01, 0xe3, 0x4e, 0xeb, 0x3a, 0x04, 0x3e, 0x5d,
+    0x52, 0xa6, 0x5e, 0x66, 0x12, 0x51, 0x88, 0xbf,
+};
+static const unsigned char tls12prf_seed[] = {
+    'k', 'e', 'y', ' ', 'e', 'x', 'p', 'a', 'n', 's', 'i', 'o', 'n',
```

Comment:
This will be incorrect on EBCDIC platforms. But do we support them still?

Reply:
Yes